### DISCUSSION TITLE
Relationship between @react-three/xr and the WebXR API

### DISCUSSION THREAD

**Discussion by ggongjukim** [pmndrs/xr#158] [Q&A] [answered]
hi, i have some question..!
As far as I know, to implement web-based augmented reality, I have to use the **WebXR API** to allow access to the device and **ThreeJS** to render virtual objects.
As such, I wonder if the idea that to implement web-based augmented reality using **React**, I should use **@react-three/fiber** instead of **ThreeJS** rendering objects and **@react-three/xr** instead of **WebXR API** is correct.

And is **@react-three/xr** a different technology independent of the **WebXR API**, or is it simply a necessary package to use the **WebXR API** to **React**?

I need a detailed explanation of **@react-three/xr** and the **WebXR API**.

Thank you

**Comment by sniok** [comment, marked as answer]
You can think of @react-three/fiber as Three.JS + React. Three.js can render XR using WebXR APIs. And @react-three/xr is just a couple of React components to make your life easier when making XR apps
